primeqa/mrc/run_mrc.py

```diff
@@ -429,36 +485,49 @@ def main():
             raise ValueError(f"Filtering on --dataset_filter_column_values ({data_args.dataset_filter_column_values}) "
                       "requires --dataset_filter_column_name to be provided.")
 
+    def process_raw_datasets(raw_datasets, max_samples, training_args, process_fn, split):
+        examples, datasets = [], []
+        for dataset in raw_datasets:
+            if max_samples is not None:
+                # We will select sample from whole data if argument is specified
+                dataset = dataset.select(range(max_samples))
+            # Feature Creation
+            with training_args.main_process_first(desc=f"{split} dataset map pre-processing"):
+                examples_ds, dataset = process_fn(dataset)
+                examples.append(examples_ds)                    
+                datasets.append(dataset)
+        return examples, datasets
     # process train data
     if training_args.do_train:
-        train_dataset = raw_datasets["train"]
-        if data_args.dataset_filter_column_values is not None:
-            logger.info(f"Filter TRAIN dataset {data_args.dataset_filter_column_name} {data_args.dataset_filter_column_values}")
-            train_dataset = train_dataset.filter(lambda example: example[data_args.dataset_filter_column_name] in (data_args.dataset_filter_column_values))
-            train_dataset = train_dataset.shuffle(seed=training_args.seed)
-            logger.info(f"Filtered TRAIN dataset size {train_dataset.num_rows}")
-        max_train_samples = data_args.max_train_samples
-        if max_train_samples is not None:
-            # We will select sample from whole data if argument is specified
-            train_dataset = train_dataset.select(range(max_train_samples))
-        # Train Feature Creation
-        with training_args.main_process_first(desc="train dataset map pre-processing"):
-            _, train_dataset = preprocessor.process_train(train_dataset)
-
+        if data_args.train_fof is not None:
+            train_datasets = raw_datasets['train']
```

Comment:
Yes. Please notice the plural in `train_datasets` -- now that run_mrc.py has to deal with multiple training datasets depending on the specification of `train_fof`, I create a `train_datasets` object for both conditions (lines 502 and 504) so that later (in line 513) they can be processed in a uniform manner.